### PR TITLE
Implement event stream content negotiation.

### PR DESCRIPTION
**Pull request by jmalloc**
#### What change does this introduce?

This PR adds support for basic content-type negotiation for event messages based on the MIME media-type.

The rules are:

1. If the client supports the existing encoding of the message, it is sent unchanged
2. Otherwise, it is re-encoded using the client's most-preferred media-type that is also supported by the server.

#### Why make this change?

This is somewhat of a future proofing measure that lets us consume events in different formats to how they are persisted in the event stream if necessary.

#### Is there anything you are unsure about?

No

#### What issues does this relate to?

None

### CODE REVIEW

**Review by danilvpetrov — APPROVED** [review]
Nice and concise PR, I like your `github.com/dogmatiq/verity/internal/x/grpcx` package methods.

File: eventstream/networkstream/server.go [hunk 1]
```diff
@@ -176,40 +183,97 @@ func (s *server) EventTypes(
...
+			failures,
+			"one or more unrecognized event types or media-types",
 		)
 	}
 
-	if len(types) == 0 {
+	return messageTypes, mediaTypes, nil
+}
+
+// transcope returns an envelope containing the message in p, transcoded into a
```
Comment (koden-km) [inline eventstream/networkstream/server.go]:
```suggestion
// transcode returns an envelope containing the message in p, transcoded into a
```

**Review by koden-km — APPROVED** [review]
Approved after the suggested typo fix.

**Comment by jmalloc** [comment]
Thanks!

**Pull request merged**